### PR TITLE
Cleanup Maven Plugin upper bounds management hacks

### PR DESCRIPTION
**Pull request by oleg-nenashev**
It is a downstream PR of https://github.com/jenkinsci/maven-plugin/pull/102, which demonstrates the cleanup of the dependency mess reported by @jtnord in [JENKINS-45271](https://issues.jenkins-ci.org/browse/JENKINS-45271)

@orrc @uhafner @reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -103,35 +109,6 @@
   </dependencies>
 
   <build>
-    <plugins>
-      <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-enforcer-plugin</artifactId>
```
Comment (oleg-nenashev) [inline pom.xml]: I hope you enjoyed it, @orrc . It was a real PITA :(

Comment (orrc) [reply pom.xml]: Yeah…

File: pom.xml [hunk 2]
```diff
@@ -26,8 +26,9 @@
     <mockito.version>1.10.8</mockito.version>
     <dashboard-view.version>2.9.4</dashboard-view.version>
     <token-macro.version>2.1</token-macro.version>
-    <jenkins-maven-plugin.version>2.17</jenkins-maven-plugin.version>
+    <jenkins-maven-plugin.version>2.18-20170921.112418-3</jenkins-maven-plugin.version>
```
Comment (jglick) [inline pom.xml]: Recommend mentioning upstream PR in a comment.

File: pom.xml [hunk 3]
```diff
@@ -26,8 +26,9 @@
...
     <matrix-project.version>1.7.1</matrix-project.version>
+    <junit-plugin.version>1.6</junit-plugin.version>
```
Comment (jglick) [inline pom.xml]: Why define a property and then use it just once? Inline.

File: pom.xml [hunk 4]
```diff
@@ -103,35 +109,6 @@
...
 
   <build>
-    <plugins>
-      <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-enforcer-plugin</artifactId>
-        <configuration>
-          <rules>
-            <requireUpperBoundDeps>
-              <excludes combine.children="append">
```
Comment (jglick) [inline pom.xml]: @vivek FYI

**Comment by uhafner** [comment]
Hmm, now analysis-test fails due to the dependency org.jenkins-ci.plugins.workflow:workflow-aggregator:1.14.2

**Comment by oleg-nenashev** [comment]
@uhafner seems I need a Jenkinsfile which would do some integration testing

**Comment by uhafner** [comment]
I will try the new version in the Christmas break. I'm currently working on a different branch (3.0) where I changed a lot of dependencies (especially workflow dependencies). It is really hard to pick the right set of libraries just to get a pipeline integration test running... (This is not just the dependencies of the maven plugin)

**Comment by uhafner** [comment]
Seems to work now, thanks :-) 

I integrated these changes in the [3.0 branch](https://github.com/jenkinsci/analysis-pom-plugin/blob/3.0/pom.xml).

**Comment by oleg-nenashev** [comment]
@uhafner Thanks a lot! It should help to make [PCT](https://github.com/jenkinsci/plugin-compat-tester) compatible with the plugins based on this POM, it gets important due to the testing of [JEP-200](https://github.com/jenkinsci/jep/tree/master/jep/200). Let me know if you need any help with doing/testing plugin upgrades.

**Pull request closed without merging**